cmd/soroban-rpc/internal/preflight/preflight.go

```diff
@@ -123,7 +123,7 @@ func GoNullTerminatedStringSlice(str **C.char) []string {
 }
 
 func GetPreflight(ctx context.Context, params PreflightParameters) (Preflight, error) {
-	hfB64, err := xdr.MarshalBase64(params.InvokeHostFunction.Function)
+	hfB64, err := xdr.MarshalBase64(params.InvokeHostFunction.Functions)
```

Comment:
I guess this implies we should change the `simulateTransaction` method to support multiple functions

Reply:
Actually, if we do it like that in Rust, it complicate the Rust<->C<->Go interface quite a bit. So, at least in that layer, we will still do it one by one.

Reply:
Also, I think that we should pass the whole operation to preflight

Reply:
i.e. 

```suggestion
	hfB64, err := xdr.MarshalBase64(params.InvokeHostFunction)
```

Reply:
That sounds like a good idea. I wasn't going to complete the implementation here, since I need your PR ready first.

Reply:
I.e. the preflightlib dependencies needs to be updated before any of these changes are meaningful.